exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporter.java

```diff
@@ -179,14 +178,29 @@ public static OtlpGrpcSpanExporter getDefault() {
   @Override
   public CompletableResultCode shutdown() {
     final CompletableResultCode result = new CompletableResultCode();
-    managedChannel.notifyWhenStateChanged(ConnectivityState.SHUTDOWN, result::succeed);
-    if (managedChannel.isShutdown()) {
+    if (managedChannel.isTerminated()) {
       return result.succeed();
     }
-    managedChannel.shutdown();
     this.spansSeen.unbind();
     this.spansExportedSuccess.unbind();
     this.spansExportedFailure.unbind();
+    managedChannel.shutdown();
+    // Remove thread creation if gRPC adds an asynchronous shutdown API.
+    // https://github.com/grpc/grpc-java/issues/8432
+    Thread thread =
+        new Thread(
```

Comment:
Shouldn't this use some kind of (background/daemon) thread factory?

Reply:
Didn't think it's needed since it's guaranteed to end in 10 (usually much less) seconds

Reply:
I think at least setting the thread as a daemon would be a good idea, so it doesn't block JVM shutdown when people don't care about this case.